src/useIntersection.ts

```diff
@@ -0,0 +1,30 @@
+import { RefObject, useEffect, useState } from 'react';
+
+const useIntersection = (
+  ref: RefObject<HTMLElement>,
+  options: IntersectionObserverInit
+): IntersectionObserverEntry | null => {
+  const [intersectionObserverEntry, setIntersectionObserverEntry] = useState<IntersectionObserverEntry | null>(null);
+
+  useEffect(() => {
+    if (ref.current) {
+      const handler = (entries: IntersectionObserverEntry[]) => {
+        setIntersectionObserverEntry(entries[0]);
+      };
+
+      const observer = new IntersectionObserver(handler, options);
+      observer.observe(ref.current);
+
+      return () => {
+        if (ref.current) {
+          observer.disconnect();
+        }
+      };
+    }
+    return () => {};
+  }, [ref.current]);
```

Comment:
It doesn't make sense to add `ref.current` to an effect dependencies for the same reason as described in #655 

Maybe we would want to add `threshold`, `root` and `rootmargin` from options to the dependency list of this effect?